### PR TITLE
Create a PatchSelectWidget

### PR DESCRIPTION
**Pull request by edg2s**
Based on TagMultiselectWidget, shows the patch number and metadata when a valid patch is entered.

### CODE REVIEW

File: PatchSelectWidget.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
+/* global OO, $ */
+
+window.PatchSelectWidget = function PatchSelectWidget( config ) {
+	var widget = this;
+
+	// Parent constructor
+	window.PatchSelectWidget.super.call( this, $.extend( {
+		allowArbitrary: true,
+		inputPosition: 'outline',
```
Comment (edg2s) [inline PatchSelectWidget.js:9]: I prefer the look of inline to outline, but also I want each task to be on a separate line so they don't jump when the names arrive from the api.

**Comment by edg2s** [comment]
This should make #196 a bit easier.

File: PatchSelectWidget.js (hunk 1 above)
Comment (edg2s) [reply PatchSelectWidget.js]: I've pushed an inline version on top of this, maybe the jumping is ok?

**Comment by edg2s** [comment]
Inline version:
image

Outline version:
image

**Comment by MatmaRex** [comment]
Looks way too professional. ;)

**Pull request merged**